src/core/reference/src/op/fake_convert.cpp

```diff
@@ -48,6 +48,7 @@ void emulate_f8e5m2_on_fp16(const float16* const arg_f, float16* out_f, size_t c
             val_bit_repr += (((rnmask > 0x0080) || (rnmask_tie == rne_tie)) << lshift);
         }
         val_bit_repr &= mask_mant; /* truncation */
+        val_bit_repr -= (((val_bit_repr & 0x7F00) == fp16_inf) << lshift); /* clamp */
```

Comment:
AFAIK, the behavior of master branch for converting different f32 values to f8e4m3 is as follows:
Value type | Convert f32 to f16 | Convert  f16 to f8e5m2
-- | -- | --
f16 overflowed value | INF | **INF (this PR change it to not-INF)**
f8e5m2 overflowed value | not-INF | **not-INF**
f8e5m2 other value | not-INF | not-INF

For overflowed values, if we apply clamp in FakeConvertDecomposition we gets not-INF, otherwise we gets INF. We can not get different behavior for "f16 overflowed value" and "f8e5m2 overflowed value".

So here, I applies this revision to make "f16 overflowed value" also get not-INF result. The other reason is to align with behavior of f8e4m3.

@dmitry-gorokhov @mitruska Please feel free to comment. Thanks a lot!

Reply:
Hello @xuchen-intel, as you mentioned the `f8e4m3` type doesn't have INF by design described in the standard, so the clamp is actually the only way to represent inf/overflow. The second type `f8e5m2` has a special sequence of bits to represent inf `S.11111.00`, so the conversion of INF in higher precision to INF in lower seems like an expected  behavior. But this is a theory, if in practice the clamp works better for the models, it could be updated.

Reply:
I agree both for the theoretical behavior and the expectation of practical models. Thanks Katarzyna!